feature-libs/product-configurator/rulebased/core/state/configurator-state-utils.spec.ts

```diff
@@ -0,0 +1,257 @@
+          'groupId@subGroupId@attributeId',
+          'attributeId'
+        )
+      ).toEqual('groupId@subGroupId');
+    });
+  });
+
+  describe('isTargeGroup', () => {
+    const groupId = 'group1';
+    const groupMatchingSupplement: Configurator.Group = {
+      id: groupId,
+      subGroups: [],
+    };
+    const groupNotMatchingSupplement: Configurator.Group = {
+      id: SUB_GROUP_ID,
+      subGroups: [],
+    };
+    const attributeSupplements: Configurator.AttributeSupplement[] = ConfiguratorTestUtils.createListOfAttributeSupplements(
+      false,
+      1,
+      0,
+      2,
+      2
+    );
+    it('should tell that a group needs to be enriched with pricing data in case group ID matches supplement key', () => {
+      expect(
+        ConfiguratorStateUtils['isTargetGroup'](
+          groupMatchingSupplement,
+          attributeSupplements
+        )
+      ).toBe(true);
+    });
+
+    it('should tell that a group does not need to be enriched if group ID does not match', () => {
+      expect(
+        ConfiguratorStateUtils['isTargetGroup'](
+          groupNotMatchingSupplement,
+          attributeSupplements
+        )
+      ).toBe(false);
+    });
+
+    it('should throw an error if supplements array is empty', () => {
+      expect(() =>
+        ConfiguratorStateUtils['isTargetGroup'](groupNotMatchingSupplement, [])
+      ).toThrowError();
+    });
+  });
+
+  describe('updateArrayElement', () => {
+    const code = 'VALUE_CODE';
+    const priceDetails: Configurator.PriceDetails = {
+      currencyIso: 'EUR',
+      value: 12,
+    };
+    let values: Configurator.Value[] = [{ valueCode: code }];
+    const valueSupplement: Configurator.ValueSupplement = {
+      attributeValueKey: code,
+      priceValue: priceDetails,
+      obsoletePriceValue: priceDetails,
+    };
+
+    it('should create new array in case predicate matches', () => {
+      const mergedValues = ConfiguratorStateUtils['updateArrayElement'](
+        values,
+        (value) => value.valueCode === valueSupplement.attributeValueKey,
+        (value) => {
+          return {
+            ...value,
+          };
+        }
+      );
+      expect(mergedValues).not.toBe(values);
+    });
+
+    it('should return same array if no match is found', () => {
+      const mergedValues = ConfiguratorStateUtils['updateArrayElement'](
+        values,
+        (value) => value.valueCode === 'NOT EXISTING',
+        (value) => {
+          return {
+            ...value,
+          };
+        }
+      );
+      expect(mergedValues).toBe(values);
+    });
+
+    it('should cope with undefined input', () => {
+      const mergedValues = ConfiguratorStateUtils['updateArrayElement'](
+        undefined,
+        (value) => value === 'NOT EXISTING',
+        (value) => {
+          return value;
+        }
+      );
+      expect(mergedValues).toBeUndefined();
+    });
+  });
+
+  describe('mergeGroupsWithSupplements', () => {
```

Comment:
I would add a test with empty attributeSupplements list.
Then we have tested both the case that the group list is empty and the supplements list is empty

Reply:
In this case an error will be thrown. Was already covered, but yes, for clarity I added this test